### PR TITLE
37 frontend generate a code to share and access room

### LINKED ISSUE #37: Frontend: generate a code to share and access room

### PR DESCRIPTION
**Pull request by ElenaGrafUZH**

### CODE REVIEW

File: src/helpers/webSocket.js [hunk 1]
```diff
@@ -5,11 +5,12 @@ import { getDomain } from "./getDomain";
 
 var stompClient = null;
 
-export const connect = async () => {
+export const connect = async (gameLink) => {
```
Comment (EriCreator) [inline src/helpers/webSocket.js:8]: We might need to create individual stomp enpoints for each game in the server.
These would look somewhat like:
registry.addEndpoint("/{gameUuid}");

The gameLink should be fetched via the rest API.

It's better not to use functions that don't exist in the backend yet..

File: src/helpers/webSocket.js [hunk 2]
```diff
@@ -38,8 +46,20 @@ export const disconnect = () => {
   }
 };
 
+export const joinRoom = (roomId) => {
+  connect();
```
Comment (EriCreator) [inline src/helpers/webSocket.js]: connect() should be used without argument, this is correct.
but you just added `gameLink` as an argument.

File: src/components/views/Game.js [hunk 3]
```diff
@@ -83,19 +89,49 @@ const Game = () => {
...
+        history.push("/board/" + lobbyId);
+      });
+    } catch (error) {
+      alert(
+        `Something went wrong while joining the lobby: \n${handleError(error)}`
+      );
+    }
+  };
+
+  const connectToWebsocket = async (lobbyId) => {
```
Comment (EriCreator) [inline src/components/views/Game.js:122]: First the connection to the lobby is made with a rest API call and then we connect to the websocket endpoint (which is not finished in the backend yet).

File: src/components/views/Game.js [hunk 4]
```diff
@@ -68,10 +70,14 @@ const Game = () => {
   const history = useHistory();
 
   const [games, setGames] = useState(mockData);
-  const [gameName, setGameName] = useState(null);
-  const [gameLink, setGameLink] = useState(null);
+  const [enterCode, setEnterCode] = useState(null);
```
Comment (EriCreator) [inline src/components/views/Game.js:73]: Code and Uuid, naming is not consistent

**Review by EriCreator — CHANGES_REQUESTED** [review]

**Review by ElenaGrafUZH — COMMENTED** [review]
Let's discuss it in person

**Review by EriCreator — APPROVED** [review]

**Pull request merged**